### PR TITLE
Adds sodium polyacrylate

### PR DESCRIPTION
**Pull request by DrSnips**
One of the common complaints I keep hearing is something about "slippery slopes" and how there isn't anything to help fix the issue of the "slippery slope."
I have created sodium polyacrylate which is a super absorbent polymer that can absorb water and lube not only in reagent containers but on floors as well. Hopefully this will un-slip the slopes.
Made with 3 parts carbon, 3 parts hydrogen, 1 part sodium, and 2 parts oxygen. This may run into issues by creating water since hydrogen and oxygen are present in the recipe but adding them in the order above will result in sodium polyacrylate. (Tested)

:cl:
 * rscadd: Added sodium polyacrylate. It absorbs water and spacelube not only in containers but on floors as well. Made with 3u carbon, 3u hydrogen, 1u sodium, and 2u oxygen.

### CODE REVIEW

**Comment by Kurfursten** [comment]
Some cursory research suggests that the synthesis of sodium polyacrylate usually involves exposure to low heat.

Maybe this would be a good basic heating recipe? Just like ten degrees above room temperature or something.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -540,6 +540,25 @@
...
+	density = 1.22
+	specheatcap = 4.14
+
+/datum/reagent/sodium_polyacrylate/reaction_turf(var/turf/simulated/T, var/volume)
+
+	if(..())
+		return 1
+
+	if(T.wet)
+		var/obj/effect/decal/cleanable/molten_item/I = new/obj/effect/decal/cleanable/molten_item(T)
```
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm]: Needs to check if there isn't already one of these decals in turf so people can't make thousands of them per tile with smoke

**Comment by DrSnips** [comment]
I could add a minor heating requirement.
EDIT: A heat requirement might cause even more issues with the formation of water since the water reaction would good off without any heat requirements and in a way prevent this recipe from happening. I will do some testing but I may need to leave it out or change the recipe to work alongside water and this might cause even more issues due it it "absorbing" extra water in the container if you have some left over if you were intending to make a large batch of this stuff.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -540,6 +540,26 @@
...
+	color = "#FFFFFF"
+	density = 1.22
+	specheatcap = 4.14
+
+/datum/reagent/sodium_polyacrylate/reaction_turf(var/turf/simulated/T, var/volume)
+
+	if(..())
+		return 1
+
+	if(T.wet)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Reagents.dm:557]: I'm just going to add a helper proc T.is_wet at this point in [16705]

**Comment by strawberrytan** [comment]
For some reason I thought bleach cleaned lube

**Comment by Exxion** [comment]
Nope
But vapor salts do

**Comment by FudgePucker** [comment]
What about giving the janitor a small bottle of it as a good luck present from their family?

**Comment by DrSnips** [comment]
I can do that in a different PR after this one. Maybe replace one of the janitor's cleaning foam grenades with a varient for drying.

**Pull request merged**